src/fsharp/PostInferenceChecks.fs

```diff
@@ -1626,11 +1705,12 @@ let CheckEntityDefn cenv env (tycon:Entity) =
     if cenv.reportErrors then 
         if not tycon.IsTypeAbbrev then 
             let typ = generalizedTyconRef (mkLocalTyconRef tycon)
-            let immediateInterfaces = GetImmediateInterfacesOfType SkipUnrefInterfaces.Yes cenv.g cenv.amap m typ
-            let interfaces = 
-              [ for ty in immediateInterfaces do
-                    yield! AllSuperTypesOfType cenv.g cenv.amap m AllowMultiIntfInstantiations.Yes ty  ]
-            CheckMultipleInterfaceInstantiations cenv interfaces m
+            let interfaces =
+                GetImmediateInterfacesOfType SkipUnrefInterfaces.Yes cenv.g cenv.amap m typ
+                |> List.collect (AllSuperTypesOfType cenv.g cenv.amap m AllowMultiIntfInstantiations.Yes)
+                |> List.filter (isInterfaceTy cenv.g)
+            let isObjExpression = false // #TODO How can I check whether it is an object expression?
```

Comment:
I looked around for any suspiciously named methods (isObjectExpression and so on), but couldn't find anything. Help appreciated.

Reply:
Afaik this code path is only hit with classes, not OEs, but not sure if that is guaranteed